service/awsconfig/v7/drainer_resource_set.go

```diff
@@ -0,0 +1,120 @@
+package v7
+
+import (
+	"context"
+
+	"github.com/cenkalti/backoff"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/operatorkit/framework"
+	"github.com/giantswarm/operatorkit/framework/context/updateallowedcontext"
+	"github.com/giantswarm/operatorkit/framework/resource/metricsresource"
+	"github.com/giantswarm/operatorkit/framework/resource/retryresource"
+
+	"github.com/giantswarm/aws-operator/client/aws"
+	"github.com/giantswarm/aws-operator/service/awsconfig/v7/key"
+)
+
+type DrainerResourceSetConfig struct {
+	GuestAWSClients aws.Clients
+	Logger          micrologger.Logger
+
+	GuestUpdateEnabled bool
+	ProjectName        string
+}
+
+func NewDrainerResourceSet(config DrainerResourceSetConfig) (*framework.ResourceSet, error) {
+	if config.GuestAWSClients.CloudFormation == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.GuestAWSClients.CloudFormation must not be empty", config)
+	}
+	if config.GuestAWSClients.EC2 == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.GuestAWSClients.EC2 must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	if config.ProjectName == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.ProjectName must not be empty", config)
+	}
+
+	var err error
+
+	//var lifecycleResource framework.Resource
+	//{
+	//	c := lifecycle.ResourceConfig{
+	//		Clients:   config.GuestAWSClients,
+	//		Logger:    config.Logger,
+	//	}
+	//
+	//	lifecycleResource, err = lifecycle.NewResource(c)
+	//	if err != nil {
+	//		return nil, microerror.Mask(err)
+	//	}
+	//}
+
+	resources := []framework.Resource{
+	//lifecycleResource,
+	}
+
+	{
+		c := retryresource.WrapConfig{
+			BackOffFactory: func() backoff.BackOff { return backoff.WithMaxTries(backoff.NewExponentialBackOff(), uint64(3)) },
+			Logger:         config.Logger,
+		}
+
+		resources, err = retryresource.Wrap(resources, c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	{
+		c := metricsresource.WrapConfig{
+			Name: config.ProjectName,
+		}
+
+		resources, err = metricsresource.Wrap(resources, c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	handlesFunc := func(obj interface{}) bool {
+		customObject, err := key.ToCustomObject(obj)
+		if err != nil {
+			return false
+		}
+
+		if key.VersionBundleVersion(customObject) == VersionBundle().Version {
```

Comment:
The cool thing is we do not need to hustle with the reconciled versions here because they are local to the package.